index.bs

```diff
@@ -7,36 +7,86 @@ Level: none
+
+<h3 id="connect-concept-section">Connect</h3>
 
-`Socket` interface {#socket-interface}
-======================================
+A socket can be constructed using a <dfn export id="connect-concept">connect</dfn> method defined in a `sockets` module (early implementations may use `vendor:sockets` for the module name), or defined on a [=binding object=].
 
-A `Socket` can be constructed using a {{connect}} method defined in a `sockets` module
-(early implementations may use `vendor:sockets` for the module name),
-or a `connect` method defined on a <a>binding object</a>.
+The connect method is the primary mechanism for creating a [=socket=] instance. It instantiates a socket with a resource identifier and some configuration values. It should syncronously return a socket instance in a <i>connected</i> state (or an error should be thrown).
+
+<h3 id="binding-object-concept">Binding Object</h3>
+
+The <dfn export>binding object</dfn> defines extra socket `connect` options. The options it contains can modify the
+behaviour of the `connect` invoked on it. Some of the options it can define:
+
+<ul>
+  <li>TLS settings</li>
+  <li>The HTTP proxy to use for the socket connection</li>
+</ul>
+
+The binding object is the primary mechanism for runtimes to introduce unique behavior for the [=connect=] method. For example, in order to support more TLS settings, a runtime may introduce a `TLSSocket` interface that extends from {{Socket}}. Thus, the binded {{connect()}} method could then utilize additional properties and configuration values that are controlled by the new `TLSSocket` interface.
+
+<pre highlight="js">
+const tls_socket = new TLSSocket({ key: '...', cert: '...' });
+tls_socket.connect();
+</pre>
+
+Additionally, the binding object does not necessarily have to be an instance of a class, nor does it even have to be JavaScript. It can be any mechanism that exposes the {{connect()}} method. Cloudflare achieves this through [environment bindings](https://developers.cloudflare.com/workers/configuration/bindings/). 
+
+<h2 id="socket-section">Socket</h2>
+
+<h3 id="using-a-socket">Using a socket</h3>
+
+<div class=example>
+A basic example of using connect with an echo server.
+  <pre highlight="js">
+  const socket = connect({ hostname: "my-url.com", port: 43 });
+
+  const writer = socket.writable.getWriter();
+  await writer.write("Hello, World!\r\n");
+
+  const reader = socket.readable.getReader();
+  const result = await reader.read();
+
+  console.log(Buffer.from(result.value).toString()); // Hello, World!
+  </pre>
+</div>
+
+<h3 id="socket-class">The {{Socket}} class</h3>
+
+The {{Socket}} class is a instance of the [=socket=] concept. It should not be instantiated directly (`new Socket()`), but instead created by calling {{connect()}}. A constructor for {{Socket}} is intentially not specified, and is left to implementors to create.
```

Comment:
```suggestion
The {{Socket}} class is an instance of the [=socket=] concept. It should not be instantiated directly (`new Socket()`), but instead created by calling {{connect()}}. A constructor for {{Socket}} is intentionally not specified, and is left to implementors to create.
```

Reply:
> A constructor for {{Socket}} is intentially not specified, and is left to implementors to create.

Do we want to leave it like this? Are there reasons why implementors might want to create a constructor for Socket? If there are then it's better for us to capture that in this document and avoid every vendor creating their own constructor.

Reply:
https://github.com/wintercg/proposal-sockets-api/issues/9